solver/src/settlement_submission.rs

```diff
@@ -31,13 +33,22 @@ pub async fn submit(
     gas_price_cap: f64,
     settlement: Settlement,
 ) -> Result<()> {
-    let nonce = transaction_count(contract)
-        .await
-        .context("failed to get transaction_count")?;
     let settlement = encode_settlement(&settlement)?;
     // Check that a simulation of the transaction works before submitting it.
     simulate_settlement(&settlement, contract).await?;
 
+    let nonce = transaction_count(contract)
+        .await
+        .context("failed to get transaction_count")?;
+    let address = match &contract.defaults().from {
+        Some(account) => account.address(),
+        None => return Err(anyhow!("no default sender address")),
+    };
+    let web3 = contract.raw_instance().web3();
+    let pending_gas_price = recover_gas_price_from_pending_transaction(&web3, &address)
```

Comment:
nit: Do we need to fail in case we cannot recover the pending transaction? Would it not be "safe" to assume there is none?

Reply:
An error in that function means communication with the node failed which is very different from no gas price so I prefer it working this way.

Reply:
I think it's good that the method returns Result<Option> just not sure an error should shortcut and fail the runloop. We could log the error and continue assuming there is no pending transaction. But we can also do that if we see this happening.

Reply:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia2.giphy.com%2Fmedia%2FQW4aFaN0GSLshdzEtA%2Fgiphy-downsized-medium.gif&token=metoo" data-gitmeme-token="metoo"><img src="https://media2.giphy.com/media/QW4aFaN0GSLshdzEtA/giphy-downsized-medium.gif" title="Created by gitme.me with /metoo"/></a>